client/lib/analytics/index.js

```diff
@@ -781,19 +781,15 @@ const analytics = {
 		},
 	},
 
-	identifyUser: function( { user = _user, newUserId = '', newUserName = '' } = {} ) {
+	identifyUser: function( newUserName = '', newUserId = '' ) {
 		const anonymousUserId = this.tracks.anonymousUserId();
 
 		// Don't identify the user if we don't have one
-		if ( ( user && user.initialized ) || ( newUserId && newUserName ) ) {
+		if ( newUserId && newUserName ) {
 			if ( anonymousUserId ) {
 				recordAliasInFloodlight();
```

Comment:
The `recordAliasInFloodlight` function should also get its user ID from reliable sources. It calls `recordParamsInFloodlightGtag` which calls `floodlightUserParams` which gets the user ID from `user.get()`. The `lib/user` dependency should be removed from `ad-tracking`, too.

This can be a follow up PR, but in the meantime, recording the alias in Floodlight (whatever it is) won't work for logged-out signup tracking. In particular, `recordSignupCompletionInFloodlight` is in danger. @jaswrks might shed more light on this. I have very little idea what I'm talking about here.

Reply:
> In particular, recordSignupCompletionInFloodlight is in danger

Reply:
@jsnajdr @ramonjd  Thanks for the heads up. I'll be taking a look at this today.

Reply:
Hi @jaswrks 👋 Did you manage to have a look at the Floodlight aliasing during signup?

Reply:
@jsnajdr Thanks for the reminder. Yes, we'll fix this along with the change noted in p4qSXL-3fA-p2#comment-11210. We also need to update this across a number of other trackers that are currently depending on `lib/user` in `ad-tracking.js`.

@jsnajdr Can you refresh my memory? What is the reason for us not updating the current state when the user ID becomes known, such that `lib/user` would continue to work?

cc @MicBosi Let's circle back to this and consider that we likely need a new method in the analytics library that returns the current user ID.

Reply:
> Can you refresh my memory? What is the reason for us not updating the current state when the user ID becomes known, such that `lib/user` would continue to work?

Because "user ID is known" is not the same thing as "user is fully logged in". To get fully logged in, i.e., get the full set of _*.wordpress.com_ session cookies, the browser needs to redirect to _wordpress.com/wp-login.php_ and then redirect back to Calypso, causing a Calypso full reload.

User ID is in `lib/user` and in Redux state only when the user is fully logged in. During signup, we are able to send authenticated REST API request immediately after the user is created, using an OAuth token, but the full login and reload happens only at the end.

It's just a convention, and in the future, it's plausible that we'll set the user ID in Redux right away. But right now, a lot of code silently assumes that "user ID is in state" equals "fully logged in" and would break if it wasn't true.

Reply:
Great, thanks for the details. I'm working on this today.

Reply:
I'm mainly focused, at the moment, on improving `identifyUser()`, such that it can update the user ID known by the analytics library. Right now it only updates Tracks. With that we can go ahead and remove the `lib/user` dependency in `ad-tracking.js`.

Reply:
Cross-referencing https://github.com/Automattic/wp-calypso/pull/37263 now in review.